sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace.py

```diff
@@ -329,8 +333,20 @@ def _from_rest_object(cls, rest_obj: RestWorkspace) -> Optional["Workspace"]:
 
         # TODO: Remove attribute check once Oct API version is out
         mlflow_tracking_uri = None
+
         if hasattr(rest_obj, "ml_flow_tracking_uri"):
-            mlflow_tracking_uri = rest_obj.ml_flow_tracking_uri
+            try:
+                from azureml.mlflow import get_mlflow_tracking_uri_v2
+
+                mlflow_tracking_uri = get_mlflow_tracking_uri_v2(rest_obj, v2_service_context)
+            except ImportError:
+                mlflow_tracking_uri = rest_obj.ml_flow_tracking_uri
+                error_msg = (
+                    "azureml.mlflow could not be imported. azureml-mlflow will not use credentials passed to `MLClient`"
+                    "Please ensure that latest 'azureml-mlflow' has been installed in the current python environment. "
```

Comment:
error message is not clear, customer can have older version of azureml-mlflow where get_mlflow_tracking_uri_v2 is not supported, called the version of azureml-mlflow in which your planning to introduce get_mlflow_tracking_uri_v2